Library/Formula/carina.rb

```diff
@@ -0,0 +1,30 @@
+class Carina < Formula
+  desc "Work with Swarm clusters on Carina"
+  homepage "https://github.com/getcarina/carina"
+  url "https://github.com/getcarina/carina.git",
+        :tag => "v0.7.1",
+        :revision => "885c4e298a9064e60a07dffd0143c38095def96e"
+  head "https://github.com/getcarina/carina.git"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+
+    carinapath = buildpath/"src/github.com/getcarina/carina"
+    carinapath.install Dir["{*,.git}"]
+
+    cd carinapath do
+      version = `git describe --tags 2> /dev/null`.strip
+      commit = `git rev-parse --verify HEAD`.strip
+
+      system "make", "get-deps"
+      system "make", "carina", "COMMIT=#{commit}", "VERSION=#{version}"
+      bin.install "carina"
+    end
+  end
+
+  test do
+    system "#{bin}/carina", "--version"
```

Comment:
This comes from looking at other formulas that are go binaries, including `docker`.

All the commands except for `--version`, `--help` and `--bash-completion` reach out to a remote service. Since this is a statically linked go binary, as long as it has been placed it's in a good spot.

I'm not completely sure what else I can test on the outside without hitting a running server.